blocks/header/header.js

```diff
@@ -85,6 +89,107 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   }
 }
 
+async function fetchSearchData() {
+  if (!window.searchData) {
+    const resp = await fetch(`/${window.location.pathname.split('/')[1]}/query-index.json`);
+    const json = await resp.json();
+    window.searchData = json;
+  }
+  return window.searchData;
+}
+
+async function search(value) {
+  const searchData = await fetchSearchData();
+  const hits = [];
+  for (let i = 0; i < searchData.data.length; i += 1) {
+    const e = searchData.data[i];
+    const text = [e.title, e.description].join(' ').toLowerCase();
+    if (text.includes(value.toLowerCase())) {
+      hits.push(e);
+    }
+  }
+  return hits;
+}
+
+function decorateSearch(block) {
+  const searchSection = block.querySelector('a > .icon-search');
+  if (searchSection) {
+    const searchElement = document.createElement('div');
+    searchElement.classList.add('nav-search');
+    searchSection.parentElement.parentElement
+      .replaceChild(searchElement, searchSection.parentElement);
+    searchElement.appendChild(searchSection);
+
+    const aside = document.createElement('aside');
+    aside.classList.add('nav-search-aside');
+    aside.classList.add('nav-search-aside-empty');
+
+    const input = document.createElement('input');
+    input.classList.add('nav-search-input');
+    input.type = 'search';
+    input.value = new URL(window.location).searchParams.get('ee_search_query');
+
+    input.addEventListener('input', async (e) => {
```

Comment:
for readability, i would extract the listener function to its own named function and just reference it here.

Reply:
Done